index.js

```diff
@@ -54,7 +54,20 @@ async function main() {
     const restored = await cache.restoreCache(cachePaths, cacheKey);
     const ret = await exec.exec('pre-commit', args, {ignoreReturnCode: push});
     if (!restored) {
-        await cache.saveCache(cachePaths, cacheKey);
+        try {
+            await cache.saveCache(cachePaths, cacheKey);
+        } catch (e) {
+            core.warning(
+                [
+                    "There was an error saving the pre-commit environments to cache:\n\n",
+                    `${e.message}\n\n`,
```

Comment:
is this a thing for all js exceptions?  I think this can be any type so it might not have a `.message`

Reply:
At least [`@actions/cache`](https://github.com/actions/toolkit/blob/af821474235d3c5e1f49cee7c6cf636abb0874c4/packages/cache/src/cache.ts#L131), throws proper errors.
But a quick browser console test, confirmed that `throw "some string"`, would work as well in js .... so that is something to watch out for, thanks for pointing it out 😄 
Quick test example:
```javascript
try{throw "foo"}catch(e){console.log(e)}
```

Reply:
`cache.saveCache` [throws an Error object](https://github.com/actions/toolkit/blob/af821474235d3c5e1f49cee7c6cf636abb0874c4/packages/cache/src/cache.ts#L170) which has the `message` property, so it should be fine.
But to be super safe this could be a ternary expression instead.
```js
e.message ? e.message : e
```

Reply:
Well better safe than sorry, or having an `undefined` in the error message.